ngraph/python/src/ngraph/opset8/ops.py

```diff
@@ -276,3 +276,57 @@ def gather(
         "batch_dims": batch_dims
     }
     return _get_node_factory_opset8().create("Gather", inputs, attributes)
+
+
+@nameable_op
+def max_pool(
+    data: NodeInput,
+    strides: List[int],
+    dilations: List[int],
+    pads_begin: List[int],
+    pads_end: List[int],
+    kernel_shape: TensorShape,
+    rounding_type: str = "floor",
+    auto_pad: Optional[str] = None,
+    index_element_type: Optional[str] = "i64",
+    axis: Optional[int] = 0,
+    name: Optional[str] = None,
+) -> Node:
+    """Perform max pooling operation and return both values and indices of the selected elements.
+
+    @param  data:                The node providing input data.
+    @param  strides:             The distance (in pixels) to slide the filter on the feature map
+                                 over the axes.
+    @param  dilations:           The dilation of filter elements(distance between elements).
+    @param  pads_begin:          The number of pixels to add at the beginning along each axis.
+    @param  pads_end:            The number of pixels to add at the end along each axis.
+    @param  kernel_shape:        The pooling operation kernel shape.
+    @param  rounding_type:       Determines used rounding schema when computing output shape.
+                                 Acceptable values are: ['floor', 'ceil']
+    @param  auto_pad:            Determines how the padding is calculated. Acceptable values:
+                                 [None, 'same_upper', 'same_lower', 'valid']
+    @param  index_element_type:  The data type used for the indices outputof this operator.
+    @param  axis:                The first dimension in the data shape used to determine the maximum
+                                 returned index value. The value is the product of all dimensions
+                                 starting at the provided axis.
+    @param  name:                The optional name for the created output node.
+
+    @return   The new node performing max pooling operation.
+    """
+    if auto_pad is None:
+        auto_pad = "explicit"
+    return _get_node_factory_opset8().create(
+        "MaxPool",
+        [as_node(data)],
+        {
+            "strides": strides,
+            "dilations": dilations,
+            "pads_begin": pads_begin,
+            "pads_end": pads_end,
+            "kernel": kernel_shape,
+            "rounding_type": rounding_type.upper(),
+            "auto_pad": auto_pad.upper(),
+            "index_element_type": index_element_type,
```

Comment:
Is `index_element_type` not depending on letters case or is it automatically casted using `lower()`/`upper()` ? If not please also specify that as in lines above.

Reply:
Good catch, it should be passed with lower case to cpp.